generators/java/generator-utils/src/main/resources/Stream.java

```diff
@@ -18,10 +26,25 @@ public final class Stream<T> implements Iterable<T> {
      */
     private final Class<T> valueType;
     /**
-     * The {@link Scanner} used for reading from the input stream and blocking when needed during iteration.
+     * The {@link Scanner} used for reading from the input stream and blocking when neede during iteration.
      */
     private final Scanner scanner;
 
+    private Options options;
+
+    private StreamType streamType;
+
+    /**
+     * The type of stream to be interpreted.
+     */
+    public enum StreamType {
+        SSE, NEWLINE
```

Comment:
nit: NDJSON might be better for NEWLINE since it's consistent with the Content-Type (e.g. [link](https://en.wikipedia.org/wiki/JSON_streaming))